src/pages/SignUp/SignUp.js

```diff
@@ -1,10 +1,354 @@
+
+            validCode: false,
+        }
+    }
+
+    validationCheck = (name, value) => {
+        if (name ==='id') {
+            let idReg = /^[a-z]+[a-z0-9]{5,19}$/g;
+            const validID = idReg.test(value);
+            this.setState({validID});
+            return;
+        } 
+        if (name ==='pw') {
+            let passwordRules = /^(?=.*[a-zA-Z])(?=.*[!@#$%^*+=-])(?=.*[0-9]).{8,16}$/;
+            const validPW = passwordRules.test(value);
+            this.setState({validPW});
+            return;   
+        } 
+        if (name === 'passwordcheck') {
+            const validDoubleCheck = this.state.pw === value
+            this.setState({validDoubleCheck});
+            return;
+        } 
+        if (name === 'name') {
+            let reg = /^[가-힣]{2,4}$/;
+            const validName = reg.test(value);
+            this.setState({validName});
+            return;
+        } 
+        if (name === 'phone') {
+            let regPhone = /^\d{3}-\d{3,4}-\d{4}$/;
+            const validPhone = regPhone.test(value);
+            this.setState({validPhone});
+            return;
+        } 
+        if (name === 'email') {
+            let emailRule = /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i;
+            const validEmail = emailRule.test(value);
+            this.setState({validEmail});
+            return;
+        } 
+    }
+
+    handleInputValueChange = (e) => {
+        const {name, value} = e.target;
+        this.validationCheck(name, value);
+        this.setState({[name]: value});
+    }
+
+    checkValidation = (e) => {
+        e.preventDefault();
+        const {id, pw, name, phone, email, passwordcheck, idDoubleCheck, validCode} = this.state;
+        if(!idDoubleCheck){
+            alert("아이디 중복확인을 해주세요")       
+            return;
+        }
+        if(!validCode){
+            alert("이메일 인증을 진행해주세요")
+            return;
+        }
+        fetch(CHECK_VALIDATION_API, {
+          method: "POST",
+          body: JSON.stringify({
+            ID: id, 
+            password: pw,
+            name: name,
+            phone_number: phone,
+            email: email,
+            passwordcheck: passwordcheck,
+          }),
+        })
+          .then((res) => res.json())
+          .then((result) => {
+
+        if (result.message === "success") {
+          alert ('회원가입이 완료되었습니다!')
+          this.props.history.push("/Login")
+        } 
+          return alert(result.message);
+        })
+    }
+    
+    handleDuplication = () => {
+        fetch(DUPLICATION_CHECK_API, {
+            method: "POST",
+            body: JSON.stringify({
+              ID: this.state.id,
+            }),
+          }).then((res) => res.json())
+            .then((result) => {
+          if (result.status === "success") {
+            this.setState({idDoubleCheck: true});
+            alert ("사용가능한 아이디입니다.");
+          } else {
+            alert(result.message);
+          }
```

Comment:
```suggestion
          if (result.status === "success") {
            this.setState({idDoubleCheck: true});
            alert ("사용가능한 아이디입니다.");
          } 
            return alert(result.message);
```
바로 return 해주시면 else문이 필요 없을 것 같습니다.

Reply:
안혜수 바보.. ㅠ...
수정하겠습니다..!